server/actions/InventoryItems.ts

```diff
@@ -0,0 +1,66 @@
+import InventoryItemSchema from 'server/models/InventoryItem'
+import * as MongoDriver from 'server/actions/MongoDriver'
+import { InventoryItem } from 'utils/types'
+import { ApiError } from 'utils/types'
+
+/**
+ * Checks to see if an item is in the inventory and will add to the quantity or
+ * create the item if needed.
+ * @param item
+ * @param quantity of item to add
+ * @returns checksIn Items to the inventory
+ */
+export async function checkInInventoryItem(
+  item: InventoryItem,
+  quantityItem: number
+) {
+  const itemMatches = await MongoDriver.findEntities(InventoryItemSchema, item)
+  if (itemMatches.length) {
+    itemMatches[0].quantity += quantityItem
+    itemMatches[0].attributes?.sort()
+    MongoDriver.updateEntity(
+      InventoryItemSchema,
+      itemMatches[0].id,
+      itemMatches[0]
+    )
+  } else {
+    const createdItem = await MongoDriver.createEntity(
+      InventoryItemSchema,
+      item
+    )
+    createdItem.attributes?.sort()
+    MongoDriver.updateEntity(InventoryItemSchema, createdItem.id, createdItem)
+  }
+}
+
+/**
+ * Checks to see if an item is in the inventory and will remove an amount or
+ * 400 error if it does not exist.
+ * @param item
```

Comment:
The format of these is `@param [parameter name] [parameter description]`. Change in all locations, and be more descriptive with what `item` is (The inventory item that is being checked out). It seems monotonous, but it's helpful in the long run. 
This should be done elsewhere as well, it's my bad that it hasn't been, but also add a `@throws ApiError when [reason for error]`
![image](https://user-images.githubusercontent.com/71574118/221768492-2788c7ae-b2c6-4b9d-8f78-228288830bef.png)